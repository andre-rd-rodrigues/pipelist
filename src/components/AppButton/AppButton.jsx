import React from "react";
import colors from "styles/_mixins.scss";
import styles from "./appbutton.module.scss";

const AppButton = (props) => {
  const { color, label } = props;
  const styling = {
    backgroundColor: !color ? "rgba(255, 255, 255, 0.5)" : colors[color],
    color: !color ? colors.dark : "white",
    border: !color && "1px solid #b0b0b0"
  };
  return (
    <button style={styling} className={styles.appButton} {...props}>
      {label}
    </button>
  );
};

export default AppButton;
