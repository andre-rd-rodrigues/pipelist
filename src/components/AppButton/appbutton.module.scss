@import "styles/mixins";

.appButton {
  @include button($green, white);
  padding: 0.5rem 1.2rem;
}
