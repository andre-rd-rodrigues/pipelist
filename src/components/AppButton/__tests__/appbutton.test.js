import { render, screen } from "@testing-library/react";
import reactTestRenderer from "react-test-renderer";
import AppButton from "../AppButton";

describe("App button", () => {
  it("should render correctly", () => {
    const wrapper = reactTestRenderer.create(<AppButton />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render according to props", () => {
    render(<AppButton label="Test" />);

    expect(screen.getByText(/test/i)).toBeInTheDocument();
  });
});
