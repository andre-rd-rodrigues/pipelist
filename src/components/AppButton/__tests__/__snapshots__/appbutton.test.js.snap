// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App button should render correctly 1`] = `
<button
  className="appButton"
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.5)",
      "border": "1px solid #b0b0b0",
      "color": undefined,
    }
  }
/>
`;
